docs/api/guidelines.md

```diff
@@ -81,6 +81,15 @@ resources which do not have deep, nested hierarchies. As a rule of thumb,
 if modifying a child resource also requires modifying the parent resource,
 you should probably make your resource higher up in the hierarchy.
 
+## Version Control
```

Comment:
This part feels a little bit weird here, as the sections are more about how to design/name endpoints and not how to call the API specifically. I added the changes lower in the document which explain how versioning works.

Reply:
It seems like we are missing a section in our API reference doc to communicate this kind of stuff (versioning, auth, common headers like `Accept`). I think that's pretty common for API reference docs. Maybe we could use https://swagger.io/specification/#info-object for this?

Otherwise we could make a separate doc and use https://swagger.io/specification/#external-documentation-object